src/code/views/menu-bar-view.js

```diff
@@ -0,0 +1,219 @@
+// TODO: This file was created by bulk-decaffeinate.
+// Sanity-check the conversion and remove this comment.
+/*
+ * decaffeinate suggestions:
+ * DS102: Remove unnecessary code created because of implicit returns
+ * DS207: Consider shorter variations of null checks
+ * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
+ */
+import DropDownView from "./dropdown-view"
+import DropDownAnchor from "./dropdown-view"
+import tr  from '../utils/translate'
+
+
+
+const {div, i, span, input} = ReactDOMFactories
+const Dropdown = createReactFactory(DropDownView)
+const {TriangleOnlyAnchor} = DropDownAnchor
+
+
+export default createReactClass({
+
+  displayName: 'MenuBar',
+
+  componentDidMount() {
+    // need to use direct DOM events because the event needs to be captured
+    if (window.addEventListener) {
+      window.addEventListener('mousedown', this.checkBlur, true)
+      window.addEventListener('touchstart', this.checkBlur, true)
+    }
+
+    return this.props.client._ui.listen(event => {
```

Comment:
Just a note for future TypeScript work.  All the _ variables were meant to be private but as you can see the `_ui` variable leaks out in various places.  It might make sense to refactor who we use it in other files and make those methods public.